### PR TITLE
Badge color, transform, and spacing fixes

### PR DESCRIPTION
**Pull request by odmln**
Light BG badges now pull in their dark counterparts for text. Orange badge uses dark text for legibility purposes. Also added `text-transform: uppercase` so you no longer have to type in all caps and some `letter-spacing` to make it read better.

### CODE REVIEW

File: src/Badge.js [hunk 1]
```diff
@@ -1,61 +1,61 @@
+import React from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
-import { space } from 'styled-system'
+import { type, space, color } from 'styled-system'
```
Comment (jxnblk) [inline src/Badge.js]: I don't think `type` is part of the styled-system API 🤔

File: src/Badge.js [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
+  letter-spacing: 0.64px;
+  ${type} ${space} ${color};
 `
 
-Badge.defaultProps = {
-  px: 2,
-  py: 1,
-  theme: theme
+const Badge = props => {
+  const badgeColor = badgeColors[props.bg] || {}
```
Comment (jxnblk) [inline src/Badge.js]: I like that this API sticks to the current convention, but it is probably worth discussing whether or not we want a color-combination-specific prop to handle both foreground and background colors

File: src/Badge.js [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
+  return <Base {...props} {...badgeColor} />
```
Comment (jxnblk) [inline src/Badge.js]: This should probably still be handled with the `type` function passed to styled-components from before. What you have here is more React-like, but less consistent with the rest of the components in the design system

File: src/Badge.js (hunk 1 above)
Comment (odmln) [reply src/Badge.js]: Yeah I don't know why i put that there 🤦

**Pull request merged**